hs-abci-test-utils/src/Tendermint/Utils/ClientUtils.hs

```diff
@@ -42,12 +40,10 @@ deliverTxEvents
   => Proxy e
   -> SynchronousResponse a b
   -> m ([Text],[e])
-deliverTxEvents pE SynchronousResponse{deliverTxResponse} =
+deliverTxEvents _ SynchronousResponse{deliverTxResponse} =
   case deliverTxResponse of
     TxResponse {txResponseEvents} ->
-      let eventName = cs $ makeEventType pE
-          es = filter ((== eventName) . eventType) txResponseEvents
-      in return . partitionEithers . map fromEvent $ es
+      return . partitionEithers . map fromEvent $ txResponseEvents
```

Comment:
@blinky3713 I don't think we can use `fromEvent` here since it's attempting to parse all the events into the same type.

Here is the an error when running the tests:
```
 test/Nameservice/Test/E2ESpec.hs:168:9:
  1) Nameservice.Test.E2E, Nameservice Spec, Can buy an existing name
       expected: []
        but got: ["Expected Event type NameClaimed does not match found Event type TransferEvent."]
```

Below is an example of the list:

```
[Event {
  eventType = "TransferEvent", eventAttributes = [KVPair {
    kVPairKey = Base64String "dHJhbnNmZXJFdmVudEFtb3VudA==", kVPairValue = Base64String "CPQD"
  }, KVPair {
    kVPairKey = Base64String "dHJhbnNmZXJFdmVudFRv", kVPairValue = Base64String "gJoxGQK9mt2ya7tWPzhwIprEqQ4="
  }, KVPair {
    kVPairKey = Base64String "dHJhbnNmZXJFdmVudEZyb20=", kVPairValue = Base64String "gJoxGQK9mt2ya7tWPzhwIprEqQ4="
  }]
}, Event {
  eventType = "NameClaimed", eventAttributes = [KVPair {
    kVPairKey = Base64String "bmFtZUNsYWltZWRPd25lcg==", kVPairValue = Base64String "gJoxGQK9mt2ya7tWPzhwIprEqQ4="
  }, KVPair {
    kVPairKey = Base64String "bmFtZUNsYWltZWROYW1l", kVPairValue = Base64String "c2F0b3NoaQ=="
  }, KVPair {
    kVPairKey = Base64String "bmFtZUNsYWltZWRWYWx1ZQ==", kVPairValue = Base64String "aGVsbG8gKGFnYWluKSB3b3JsZA=="
  }, KVPair {
    kVPairKey = Base64String "bmFtZUNsYWltZWRCaWQ=", kVPairValue = Base64String "CPQD"
  }]
}]
```
I suggest we handle the parsing in the individual tests and change `deliverTxEvents`'s return type from: ` m ([Text],[e])` to  `m ([Text],[Event])`